core/domain/exp_domain.py

```
@@ -3172,17 +3175,27 @@ def serialize(self) -> str:
         # files must add a version parameter to their files with the correct
         # version of this object. The line below must then be moved to
         # to_dict().
-        # The dictionary returned by `to_dict()` method is ExplorationDict
-        # and ExplorationDict does not contain `version`, `created_on` and
-        # `last_updated` keys, but here we are defining those keys which
-        # causes MyPy to throw error TypedDict has no `version` key. Thus
-        # to silent the error, we used ignore here.
+        # Here we use MyPy ignore because the dictionary returned by `to_dict()`
```

In cases like this, define a new dict and name/type it appropriately.

I think in general, if your stated reasoning is "to silence the error", it is better to think again. (You don't want other coders to follow that logic, right?)

---

Done